R/bundlePackage.R

```diff
@@ -25,135 +25,59 @@ bundlePackages <- function(appDir,
   )
   deps <- snapshotRDependencies(appDir, inferredRDependencies, verbose = verbose)
 
-  packageMessages <- c()
-  errorMessages <- c()
-  packages <- list()
-
-  # construct package list from dependencies
-  for (i in seq_len(nrow(deps))) {
-    name <- deps[i, "Package"]
-
-    # get package info
-    info <- as.list(deps[i, c("Source", "Repository")])
-
-    # include github package info
-    info <- c(info, as.list(deps[i, grep("Github", colnames(deps), perl = TRUE, value = TRUE)]))
-
-    # get package description; note that we need to remove the
-    # packageDescription S3 class from the object or jsonlite will refuse to
-    # serialize it when building the manifest JSON
-    # TODO: should we get description from packrat/desc folder?
-    info$description <- suppressWarnings(unclass(utils::packageDescription(name)))
-
-    # TODO(HW): should this code be removed? I don't see how it is ever reached
-    # if description is NA, application dependency may not be installed
-    if (is.na(info$description[1])) {
-      errorMessages <- c(
-        errorMessages,
-        paste0(
-          capitalize(assetTypeName), " depends on package \"", name, "\"",
-          "but it is not installed. Please resolve before continuing."
-        )
-      )
-      next
-    }
-
-    # validate package source (returns a message if there is a problem)
-    packageMessages <- c(packageMessages, validatePackageSource(deps[i, ]))
-
-    # good to go
-    packages[[name]] <- info
+  not_installed <- !vapply(deps$Package, is_installed, logical(1))
```

Comment:
It's going to take me time to grok that things like `is_installed` are coming from `rlang` rather than part of this package. All other external references use `packagename::XX` names.

Reply:
Yeah, our philosophy is that rlang is the sort of missing stdlib, that we've decided that we can't live with out and use so frequently that `::` is high friction.